accounts/views.py

```diff
@@ -101,13 +115,185 @@ def signup(request):
         return render(request, "accounts/login.html", {"form": form})
 
 
+def onboarding(request):
+    # TODO uncomment this and make else redirect to page not found or home
+    # if(request.session.get("onboarding")):
+    return render(
+        request,
+        "accounts/onboarding.html",
+    )
+
+
+def onboarding_genre_view(request):
+    """
+    This function handles the POST/GET requests for the onboarding genre form.
+    """
+    if not request.user.is_authenticated:
+        return redirect("accounts:login")
+
+    if request.method == "POST":
```

Comment:
i honestly think that this would look way better in a helper function
or at the very least, some newlines bro